src/algebra/poly/poly.rs

```diff
@@ -0,0 +1,415 @@
+            coeffs.push(lhs_coeff + rhs_coeff);
+        }
+
+        DensePolynomialResult::from_coeffs(coeffs)
+    }
+}
+impl_borrow_variants!(DensePolynomialResult<C>, Add, add, +, DensePolynomial<C::ScalarField>, C: CurveGroup);
+impl_commutative!(DensePolynomialResult<C>, Add, add, +, DensePolynomial<C::ScalarField>, C: CurveGroup);
+
+impl<C: CurveGroup> Add<&DensePolynomialResult<C>> for &DensePolynomialResult<C> {
+    type Output = DensePolynomialResult<C>;
+    fn add(self, rhs: &DensePolynomialResult<C>) -> Self::Output {
+        // We do not pad the coefficients here, it requires fewer gates if we avoid padding
+        let mut coeffs = Vec::new();
+        let (shorter, longer) = if self.coeffs.len() < rhs.coeffs.len() {
+            (&self.coeffs, &rhs.coeffs)
+        } else {
+            (&rhs.coeffs, &self.coeffs)
+        };
+
+        for (i, longer_coeff) in longer.iter().enumerate() {
+            let new_coeff = if i < shorter.len() {
+                &shorter[i] + longer_coeff
+            } else {
+                longer_coeff.clone()
+            };
+
+            coeffs.push(new_coeff);
+        }
+
+        DensePolynomialResult::from_coeffs(coeffs)
+    }
+}
+impl_borrow_variants!(DensePolynomialResult<C>, Add, add, +, DensePolynomialResult<C>, C: CurveGroup);
+
+// --- Negation --- //
+impl<C: CurveGroup> Neg for &DensePolynomialResult<C> {
+    type Output = DensePolynomialResult<C>;
+    fn neg(self) -> Self::Output {
+        let mut coeffs = Vec::with_capacity(self.coeffs.len());
+        for coeff in self.coeffs.iter() {
+            coeffs.push(-coeff);
+        }
+
+        DensePolynomialResult::from_coeffs(coeffs)
+    }
+}
+impl_borrow_variants!(DensePolynomialResult<C>, Neg, neg, -, C: CurveGroup);
+
+// --- Subtraction --- //
+#[allow(clippy::suspicious_arithmetic_impl)]
+impl<C: CurveGroup> Sub<&DensePolynomial<C::ScalarField>> for &DensePolynomialResult<C> {
+    type Output = DensePolynomialResult<C>;
+    fn sub(self, rhs: &DensePolynomial<C::ScalarField>) -> Self::Output {
+        let negated_rhs_coeffs = rhs.coeffs.iter().map(|coeff| -(*coeff)).collect();
+        let negated_rhs = DensePolynomial::from_coefficients_vec(negated_rhs_coeffs);
+
+        self + negated_rhs
+    }
+}
+impl_borrow_variants!(DensePolynomialResult<C>, Sub, sub, -, DensePolynomial<C::ScalarField>, C: CurveGroup);
+
+#[allow(clippy::suspicious_arithmetic_impl)]
+impl<C: CurveGroup> Sub<&DensePolynomialResult<C>> for &DensePolynomialResult<C> {
+    type Output = DensePolynomialResult<C>;
+    fn sub(self, rhs: &DensePolynomialResult<C>) -> Self::Output {
+        // Negate the rhs then use the `Add` impl
+        self + (-rhs)
+    }
+}
+
+// --- Multiplication --- //
+
+// TODO: For each of the following implementations, we can await all coefficients to be available
+// and then perform an FFT-based polynomial multiplication. This is left as an optimization
```

Comment:
This is obviously an optimization for down the line, but if and when we do implement this, I think it would be neat to opt in to this optimization, and also have polynomial multiplication defined over point-form. Where I'm coming from, is that I believe that for some of the polynomial multiplications in the Plonk prover algorithm, we should already have both point and coefficient forms for the multiplicands (for example, when calculating $a(X)b(X)q_M(X)$ ), and so we wouldn't need to invoke an extra FFT here, right?

At any rate, just rambling, nothing actionable for this PR

Reply:
Yeah we could have a point representation if we like, I'd say that is more of interest to the calling code, i.e. `mpc-jellyfish` to construct such an `ifft` based multiplication